### PR TITLE
Feature/get max values

### PR DESCRIPTION
**Pull request by martinyde**
#### Link to ticket

https://leantime.itkdev.dk/#/tickets/showTicket/3122

#### Description

  - Add max values calculation.
  - Add track tool sum calculation
  - Add track calculation

### CODE REVIEW

File: web/modules/custom/ai_screening_project_track/src/Helper/ProjectTrackTypeHelper.php [hunk 1]
```diff
@@ -171,6 +173,36 @@ public function validateConfiguration(array $configuration): void {
...
+      // Look at each field in the webform.
+      foreach ($elements as $element) {
+        // Currently we only look at weighted radios.
+        if ('ai_screening_weighted_radios' === $element['#type'] && array_key_exists('#options', $element)) {
+          // Get all options for the field.
+          foreach ($element['#options'] as $optionValues => $option) {
+            $values = FormHelper::getIntegers($optionValues);
+            // Get possible max for each element.
+            foreach ($values as $key => $value) {
+              $currentMax[$key] = max($currentMax[$key] ?? 0, $value);
```
Comment (rimi-itk) [inline web/modules/custom/ai_screening_project_track/src/Helper/ProjectTrackTypeHelper.php:196]: Shouldn't the values be accumulated, i.e. added together, to get the maximum value for each dimension? If I understand the code correctly, you just get the largest value of any option.

If we have two options, e.g.

```
1,2,4: Yes
2,3,1: No
```

, then I expect the maximum values to be `3,5,5` and not `2,3,4`. I may be missing something …

File: web/modules/custom/ai_screening_project_track/src/Computer/WebformSubmissionProjectTrackToolComputer.php [hunk 2]
```diff
@@ -25,6 +27,36 @@ public function supports(ProjectTrackToolInterface $tool, EntityInterface $entit
...
+        // expect this to happen on all non-radio fields we don't take further
+        // actions.
+      }
+    }
+
+    $mappedValues = array_map(function ($key, $value) {
+      return [$key, $value];
+    },
+      $dimensions, $calculated
+    );
```
Comment (rimi-itk) [inline web/modules/custom/ai_screening_project_track/src/Computer/WebformSubmissionProjectTrackToolComputer.php:53]: Why do you map to a list of “index“,  “value”, e.g.

``` php
[
 [0, 87],
 [1, 42],
]
```

I have a feeling that just using `$calculated`, i.e. `[0 => 87, 1 => 42]`, may be easier further on.

**Review by rimi-itk — COMMENTED** [review]
This looks good. However, I have some questions on what is actually going …

File: web/modules/custom/ai_screening_project_track/src/Helper/ProjectTrackTypeHelper.php (hunk 1 above)
Comment (martinyde) [reply web/modules/custom/ai_screening_project_track/src/Helper/ProjectTrackTypeHelper.php:197]: Not quite we don't sum each options value but rather the highest valued option of each question so:

```
**Question1**
1,2,4: Yes
2,3,1: No

**Question2**
4,5,6: Full
3,4,7: Half 
8,4,1: Quarter
4,7,3: Double
```

We should expect ```10,10,11```

File: web/modules/custom/ai_screening_project_track/src/Helper/ProjectTrackHelper.php [hunk 3]
```diff
@@ -56,19 +85,21 @@ public function __construct(
    * @see self::hasTrackData()
    */
   public function getToolData(ProjectTrackInterface $track, ?string $key = NULL): mixed {
```
[the remark this answers is not in the document]
Comment (martinyde) [reply web/modules/custom/ai_screening_project_track/src/Helper/ProjectTrackHelper.php:87]: Changed

File: web/modules/custom/ai_screening_project_track/src/Computer/WebformSubmissionProjectTrackToolComputer.php (hunk 2 above)
Comment (martinyde) [reply web/modules/custom/ai_screening_project_track/src/Computer/WebformSubmissionProjectTrackToolComputer.php]: Changed this

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**